src/core/thread/mle.cpp

```diff
@@ -2835,6 +2842,11 @@ otError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageInfo
         otLogWarnMleErr(error, "Failed to process Data Response");
     }
 
+    if (mDataRequestState == kDataRequestNone && !IsRxOnWhenIdle())
+    {
+        Get<DataPollManager>().StopFastPolls();
```

Comment:
There may be another message sent that is waiting for a response, so we should not stop fast polling on receiving a response. We can possibly have multiple MLE Data Req messages already sent. We have mechanism in `HandleMessageTransmissionTimer` to retry them a bunch of time.